transaction/simple.go

```diff
@@ -2,29 +2,35 @@ package transaction
 
 import (
 	"github.com/gluster/glusterd2/context"
+
+	"github.com/pborman/uuid"
 )
 
 // SimpleTxn is transaction with fixed stage, commit and store steps
 type SimpleTxn struct {
 	// Ctx is the transaction context
 	Ctx *context.Context
 	// Nodes are the nodes where the stage and commit functions are performed
-	Nodes []string
+	Nodes []uuid.UUID
 	// LockKey is the key to be locked
 	LockKey string
 
+	// Stage is the registered name of the staging StepFunc
 	// Stage function verifies if the node can perform an operation.
-	Stage StepFunc
+	Stage string
+	// Commit is the registered name of the commit StepFunc
 	// Commit performs the operation on the a node
-	Commit StepFunc
+	Commit string
+	// Store is the registered name of the store StepFunc
 	// Store stores the results of an operation. This will only be run on the leader
-	Store StepFunc
+	Store string
```

Comment:
On failure store, what will be the action ? Would commit rollback be initiated ?